modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/utils/SW360ReleaseAdapterUtils.java

```diff
@@ -72,18 +72,13 @@ private static void setCPEId(SW360Release release, Artifact artifact) {
                 .ifPresent(release::setCpeId);
     }
 
-    private static void setCoordinates(SW360Release release, Artifact artifact) {
-        release.setCoordinates(getMapOfCoordinates(artifact));
-    }
+    private static void setPackageUrl(SW360Release sw360Release, Artifact artifact) {
+        Optional<PackageURL> packageURL = artifact.askForAll(ArtifactCoordinates.class)
```

Comment:
Here, I think this findAny is not the adequate solution. In ArtifactUtils there is a default preference for coordinates which returns the coordinates according to priority, this seems to be the right thing to do.

If I remember right, we discussed that we want to store all known purls for the object. In the same way as the hash codes, i.e., with an _<number> suffix.

Reply:
I had a call yesterday with @sschuberth. In our discussion we came to the conclusion that per artifact there should only be one purl. Did we get that wrong?

Reply:
This is the principle discussion whether we have one data entry for a component with different established distributions or several. E.g. a repackaged maven component, which is provided in a p2 repository has two different coordinates with the same content.

One can argue that these are two components and therefore two instances with different purls, but effectively the metadata for the component are the same, as the sources from which the component has been build are the same.

So, my preferred solution is to have all known instances of a component managed in one data object. That means multiple purls pointing to the same data object

A similar case is in nuget, where we have the original source in nuget.org, but the same  component, differently packaged from Microsoft, or with alternative package managers in other languages. In SW360, we should be able to manage multiple instances, whereas I think the way we did it in Antenna with multiple coordinates might be wrong because we look at a given set of instances, not the overall data of a component.

Adding @sschuberth into the discussion

Reply:
> the metadata for the component are the same, as the sources from which the component has been build are the same.

IMO we can and should use PURL to make exactly that distinction: To differentiate between different packagings of the same component or source code (e.g. `.tar.gz` vs. `.zip`), and to differentiate between different locations from which the same package (e.g. Maven Central vs. JCenter).

Linking some (semi-)related issue discussions here:

"separate type from provider": https://github.com/package-url/purl-spec/issues/33
"Source of repackaged code": https://github.com/clearlydefined/curated-data/issues/25

Reply:
on the SW360 we do not distinguish between an artifact, its sources, or whether it was pulled from maven or from github. This means that in SW360 all the corresponding PURLs should be added to one release.

I think two artifacts with the same name and version are the same item in SW360.

Reply:
@maxhbr  what format would you suggest for the case: two purls from the same type?
(like started in this conversation: https://github.com/eclipse/antenna/pull/291#discussion_r326483216)

Reply:
Going over the discussion between you @sschuberth and Jeff, it seems that I have a different understanding of what a purl references. From what I see, a purl references a unique id/coordinates for a component release unifying the different ways of how component coordinates are represented between technologies.

That means, that for one technology (e.g. maven) I would expect exactly one purl that references a version of a component. Having said that, a technology means basically a packaging type or a package manager.

There are several copnsequences:
1. A component release available in different package manager technologies has multiple purls, each of which uniquely identifies the component release.
2. I do not care about the provider or source of the component, if it is the content, it has the corresponding purl of the package manager technology, since both the instances are the same thing, i.e., built from the same source code.
3. If the content is different, then it is a different release, i.e., it has a different purl
4. Whether I trist that the release has the content the purl indicates is not a matter of purl but a matter of do I trust the source of this release.
5. The source of the release is something I would like to know but this is our of scope of the purl spec.
6. The purl type indentifies the technology with which I use the component in the build, aka as the package manager technology. I still can download the component from the source and build it on my own, but even then the normal way would be to build the corresponding library and use it as if I would download it directly from the internet: e.g., Build a Jar, deploy it to a maven repo manager (e.g. Nexus) and reference it with the repo manager as provider in my pom.xml.
7. If I do not use it as described in 6, e.g., if I download the source and directly build it together with my sources, this still is the same content, but from my point, I would need to use a different purl, e.g., with type github, because the way I use it is different. I do not use the component as Maven component anymore, but as bunch of sources, so different purl.

I think that covers my line of thinking.

Reply:
Specifically for this PR and how to implement the solution, what is the implication of your line of thinking for this implementation?

Reply:
- We have multiple purls per release
- The purls should be requested as proposed in the ArtifactUtils default sequence, not findAny or findFirst
- Since I assume only one purl per technology, I would follow Max comment from below that the key in the external id should be the type used in the purl as redundant element
- Purls are used in the Release object with PackageURL class not string and only when trnaslated into the JSON the string representation is used.